src/uk/gov/hmcts/contino/MetricsPublisher.groovy

```diff
@@ -66,19 +65,26 @@ class MetricsPublisher implements Serializable {
     ]
   }
 
-  @NonCPS
   private def createDocument(metrics, cosmosDbUrl) {
-    def client = new DocumentClient(cosmosDbUrl, env.COSMOSDB_TOKEN_KEY, null, null)
+    steps.sh "mkdir -p /tmp/metrics-reporting"
 
-    try {
-      def data = JsonOutput.toJson(metrics).toString()
-      client.createDocument(resourceLink, new Document(data), null, false)
-      data = null
-    } finally {
-      client.close()
-      client = null
-    }
+    steps.writeFile(file: '/tmp/metrics-reporting/package.json', text: steps.libraryResource("${METRICS_RESOURCE_PATH}/package.json"))
+    steps.writeFile(file: '/tmp/metrics-reporting/yarn.lock', text: steps.libraryResource("${METRICS_RESOURCE_PATH}/yarn.lock"))
+    steps.writeFile(file: '/tmp/metrics-reporting/metrics-publisher.js', text: steps.libraryResource("${METRICS_RESOURCE_PATH}/metrics-publisher.js"))
+
+    def data = JsonOutput.toJson(metrics).toString()
+    steps.sh """
+      cd /tmp/metrics-reporting/
+      chmod +x metrics-publisher.js
+      yarn install
+
+      export COSMOS_DB_URL=${cosmosDbUrl}
+      export COSMOSDB_TOKEN_KEY=${env.COSMOSDB_TOKEN_KEY}
+      export COSMOS_COLLECTION_ID=pipeline-metrics
 
+      ./metrics-publisher.js '${data}'
+    """
+    steps.echo "Published metrics"
```

Comment:
What's the performance like? This gets run on every stage of every build

Reply:
takes one second or less,
https://sandbox-build-beta.platform.hmcts.net/job/timj-test/17/console